src/shared/quota-util.c

```diff
@@ -34,3 +35,49 @@ int quotactl_fd_with_fallback(int fd, int cmd, int id, void *addr) {
 
         return RET_NERRNO(quotactl(cmd, devnode, id, addr));
 }
+
+int quota_query_proj_id(int fd, uint32_t proj_id, struct dqblk *req) {
+        int r;
+
+        assert(fd >= 0);
+        assert(req);
+
+        zero(*req);
```

Comment:
is this necessay?

we generally have this rule in our codebase that return parameters should not be touched on failure, and should be fulle initialized on success. this touch here interferes with that, because we'll touch the structure even though the func can still fail.

if this is not necessary, just drop it?

if it is necessary to pass a zero initialized buffer in (which would surprise me), then use a local variable for this, and only copy over once the func cannot fail anymore